lib/temporal/testing/local_workflow_context.rb

```diff
@@ -53,13 +53,20 @@ def execute_activity(activity_class, *input, **args)
         )
         context = LocalActivityContext.new(metadata)
 
-        result = activity_class.execute_in_context(context, input)
-
-        if context.async?
-          execution.register_future(context.async_token, future)
+        begin
+          result = activity_class.execute_in_context(context, input)
+        rescue StandardError => e
+          # Capture any failure from running the activity into the future
+          # instead of raising immediately. This better simulates the behavior
```

Comment:
super nit: rm better 
(mentioning due to interesting meta-rule of not putting "what's changing" language into code comments)

Reply:
Fixed